metapool/prep.py

```diff
@@ -842,3 +843,84 @@ def qiita_scrub_name(name):
         the sample name, formatted for qiita
     """
     return re.sub(r'[^0-9a-zA-Z\-\.]+', '.', name)
+
+
+def _organize_by_project(pre_prep):
+    # extract project-level metadata
+
+    gb = pre_prep.groupby('project_name')
+
+    results = [gb.get_group(grp) for grp in gb.groups]
+
+    projects = []
+    data = {}
+
+    for project in results:
+        # the set of unique project_names should only be one value
+        project_name = set(project['project_name']).pop()
+
+        # assert that all values in 'contains_replicates' column for a given
+        # project are either all True or all False. Handle case-variations.
+        values = set([x.lower() == 'true' for x in
+                      list(project['contains_replicates'])])
+        if len(values) != 1:
+            raise ValueError(f"values in 'contains_replicates' column contain "
+                             "values for both True and False for project "
+                             f"'{project_name}'")
+
+        projects.append((project_name, values.pop()))
```

Comment:
why `pop` vs. `[0]`?

Reply:
the way I'm doing it, values is a set() so it won't accept indices. I can index it with your version.